### PR TITLE
Location marker

### PR DESCRIPTION
**Pull request by johan-t**

### CODE REVIEW

File: src/components/LocationMarker/LocationMarker.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        const intervalId = setInterval(getPosition, 2000);
+
+        // Clear interval on component unmount
+        return () => clearInterval(intervalId);
+    }, []);
+
+    return (
+        <div>
+            {position && (
+                <Marker position={position} icon={LocationIcon} >
```
Comment (johan-t) [inline src/components/LocationMarker/LocationMarker.tsx:41]: Use a css black dot instead of the image

File: src/components/Markers/Markers.tsx [hunk 2]
```diff
@@ -1,34 +1,82 @@
...
+        data.map((item, index) => {
+          const station = item.station.name || '';
+          const line = item.line;
+          const direction = item.direction.name || '';
+
+            const latitude = item.station.coordinates.latitude;
+            const longitude =  item.station.coordinates.longitude;
+
+            return (
+              <Marker key={`${line}-${index}`} position={[latitude, longitude]} icon={MarkerIcon}>
```
Comment (johan-t) [inline src/components/Markers/Markers.tsx:68]: Use a css red dot instead of the icon

File: src/functions/dbUtils.tsx [hunk 3]
```diff
@@ -1,10 +1,13 @@
-export function getCoordinates(setData: React.Dispatch<React.SetStateAction<Array<[number, number, string]>>>) {
-    fetch('/data')
+import { MarkerData } from '../components/Markers/Markers';
+
+export function getLatestData(setData: React.Dispatch<React.SetStateAction<MarkerData[]>>) {
```
Comment (johan-t) [inline src/functions/dbUtils.tsx:3]: name the function something more descriptive like 'getLatestTicketInspectorInfo'

basically keep it in sync with the endpoint name so "'get'+{descriptiveEndpointName}"

**Review by johan-t — COMMENTED** [review]
Good work

File: public/index.html [hunk 4]
```diff
@@ -3,7 +3,7 @@
   <head>
     <meta charset="utf-8" />
     <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
-    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no" />
```
Comment (brandesdavid) [inline public/index.html:6]: This optimizes the mobile user flow.
Sets the content to the whole device width. Scales to 100% and makes the user not scale the page, but the map.

File: src/components/LocationMarker/LocationMarker.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import L from 'leaflet';
+import React, { useEffect, useState } from 'react';
+import { Marker, Popup } from 'react-leaflet';
+import { createLocationMarkerHTML } from '../../functions/mapUtils';
+
+const LocationMarker = () => {
+    const [position, setPosition] = useState<[number, number] | null>(null);
+
+        const LocationIcon = L.divIcon({
```
Comment (brandesdavid) [inline src/components/LocationMarker/LocationMarker.tsx:9]: This adds the css icons for the markers (take a look at mapUtils.tsx in functions folder)

File: src/components/Map/Map.tsx [hunk 6]
```diff
@@ -11,13 +12,16 @@ interface MapProps {
 const Map: React.FC<MapProps> = ({ formSubmitted }) => {
     // Berlin as Standardview
     const position: LatLngTuple = [52.5162,13.3880];
-    
+
+    const maxBounds = latLngBounds([52.96125019866001, 12.509131386425151], [52.014679000584486, 14.382300343810543]);
```
Comment (brandesdavid) [inline src/components/Map/Map.tsx:16]: MaxBounds so you cant scale to other countries or very far outside berlin

File: src/components/Map/Map.tsx [hunk 7]
```diff
@@ -11,13 +12,16 @@ interface MapProps {
...
+
   return (
-        <MapContainer id='map' center={position} zoom={13} scrollWheelZoom={true}>
+        <MapContainer id='map' center={position} zoom={13} scrollWheelZoom={true} maxBounds={maxBounds}>
         <TileLayer
             attribution= '<a href="https://jawg.io" title="Tiles Courtesy of Jawg Maps" target="_blank">&copy; <b>Jawg</b>Maps</a>'
             url={`https://tile.jawg.io/jawg-streets/{z}/{x}/{y}{r}.png?access-token=${process.env.REACT_APP_JAWG_ACCESS_TOKEN}`}
         />
+        <LocationMarker />
```
Comment (brandesdavid) [inline src/components/Map/Map.tsx:24]: The location marker

File: src/components/Markers/Markers.tsx [hunk 8]
```diff
@@ -1,34 +1,87 @@
...
+import L from 'leaflet';
+import { getRecentTicketInspectorInfo } from '../../functions/dbUtils';
+import { createMarkerHTML } from '../../functions/mapUtils';
 
 interface MarkersProps {
-    formSubmitted: boolean;
+  formSubmitted: boolean;
 }
 
+export type MarkerData = {
```
Comment (brandesdavid) [inline src/components/Markers/Markers.tsx:12]: This is the received marker data we get from the response, also using it directly for the map

File: src/components/Markers/Markers.tsx [hunk 9]
```diff
@@ -1,34 +1,87 @@
...
+    const MarkerIcon = L.divIcon({
+        className: 'custom-icon',
+        html: createMarkerHTML(),
+        iconSize: [20, 20]
+    });
+
+ useEffect(() => {
+    const fetchData = async () => {
+        const newData = await getRecentTicketInspectorInfo();
+        if (JSON.stringify(newData) !== JSON.stringify(data)) {
```
Comment (brandesdavid) [inline src/components/Markers/Markers.tsx:45]: This is important, everytime i used the app on my phone, it always refreshed every 5 seconds and made the dots and popups disappear. Now with this, it only changes the map when data has been changed

File: src/functions/dbUtils.tsx [hunk 10]
```diff
@@ -1,10 +1,14 @@
-export function getCoordinates(setData: React.Dispatch<React.SetStateAction<Array<[number, number, string]>>>) {
-    fetch('/data')
-    .then(response => response.json())
-    .then(data => {
-        setData(data);
-    })
-    .catch(error => console.error(error));
+import { MarkerData } from '../components/Markers/Markers';
+
+export async function getRecentTicketInspectorInfo(): Promise<MarkerData[]> {
```
Comment (brandesdavid) [inline src/functions/dbUtils.tsx:3]: changed name to the backend name

File: src/functions/mapUtils.tsx [hunk 11]
```diff
@@ -0,0 +1,20 @@
+export const createLocationMarkerHTML = () => {
```
Comment (brandesdavid) [inline src/functions/mapUtils.tsx:1]: These are the marker html css divs

**Review by brandesdavid — APPROVED** [review]
@johan-t take a look at my comments in the review and the two last commit messages

**Pull request merged**